src/main/scala/projectEuler/LargestPrimeFactorCalculator.scala

```diff
@@ -0,0 +1,26 @@
+package projectEuler
+
+import scala.math._
+
+/**
+ * Problem 3. Largest prime factor
+ * The prime factors of 13195 are 5, 7, 13 and 29.
+ * What is the largest prime factor of the number 600851475143 ?
+ * @see https://projecteuler.net/problem=3
+ */
+class LargestPrimeFactorCalculator(num: Long) {
+
+  private def isFactor(numerator: Long)(denominator: Long): Boolean = {
```

Comment:
Using currying is a bit overcomplicated for this scenario - it should only be used when the curried function needs to be passed around. I think you have use currying so that you can use this function as a predicate on line 18, but this can be solved in a different way